creating_citation_counts_tsv/scripts/summarize_gene_citations_all_species.py

```diff
@@ -2,270 +2,215 @@
-    
-    # Outputing the first element in the tax_id variable
-    print(tax_id)
+    # Getting the taxonomy ID for the species and saving it to the species_tax_id variable
+    species_tax_id = first_dict_entry["tax_id"]
 
-    # Getting the taxonomy information for the taxonomy ID given and saving it to the taxonomy variable
-    taxonomy = (sc.textFile("creating_citation_counts_tsv/taxonomy_name")
-                        .filter(lambda x: x[0] != '#')
-                        .map(lambda x: x.split('\t'))
-                        .filter(lambda x: int(tax_id) == int(x[0]) )
-                        .filter(lambda x: "scientific name" in x[6] )
-                        .map(lambda x: ((str(x[0]), str(x[2]), str(x[6]))))
-                     )
-    
-    # Getting the taxonomy name from the taxonomy information and saving it to the tax_name variable
-    tax_name = taxonomy.collect()[0][1]
-    
-    # Outputing the tax_name variable
-    print(tax_name)
+    # Outputing the species_tax_id variable
+    print("species_tax_id: " + species_tax_id)
 
-    # Returning the gene_info and tax_name variable
-    return gene_info, tax_name
+    # Getting the taxonomy scientific name for the taxonomy ID given and saving it to the taxonomy_scientific_name variable
+    taxonomy_scientific_name = ""
+    # This file has no headers so here's an example instead:
+    #9606    |       Homo sapiens    |               |       scientific name |
+    with open(taxonomy_name_tsv) as fd:
+        rd = csv.reader(fd, delimiter='\t')
+        for row in rd:
+            row_tax_id = str(row[0])
+            name = str(row[2]) # the file does not include column headers so these are guesses based on the data
+            name_type = str(row[6])
+            if row_tax_id==species_tax_id and name_type=="scientific name":
+                taxonomy_scientific_name = name
+                break
 
-def get_ref_gene(species, gene_info): 
+    # Outputing the taxonomy_scientific_name variable
+    print("taxonomy_scientific_name: " + taxonomy_scientific_name)
+
+    # Returning the gene_info and taxonomy_scientific_name variable
+    return gene_symbol__gene_info__dict, taxonomy_scientific_name
+
+def get_ref_gene(species, gene_symbol__gene_info__dict): 
     """ 
     Getting a list of the gene reference information per gene.
     """
-    
-    # Going through all the files in the species directory 
-    for refGene_file in os.listdir(f"creating_citation_counts_tsv/data/{species}"):
-         #Since the reference file is uniquely named and could not be hard-coded, this will filter out "gene_info" and "gene2pubmed" to read the reference file
-        if refGene_file != "gene_info" and "gene2pubmed":
-            # Adding gene's reference information to the list created from getting_gene_info per gene and saving it to the refGene variable
-            refGene = (sc.textFile(f"creating_citation_counts_tsv/data/{species}/{refGene_file}")
```

Comment:
maybe a python versioning issue, but this was reading all 3 files. this bug was fixed